### ISSUE TITLE
Support nextjs vercel deployment

### ISSUE THREAD

**Issue by ItayTur** [wix/stylable#1710]
I work on nextjs project integrated with stylable.

I followed this [issue](https://github.com/wix/stylable/issues/1212), and it works locally.

the problem is the deployment.

when deploying using `vercel` the build time reach it's max (45min), and fail because of it.

I maid some progress, and change the `build command` in the `package.json`, to `next build` and the build completed, but no css was loaded. 

the browser throw 404 errors on the `st.css` files.

Build logs show this message: 

> Loading a Stylable stylesheet via webpack loaders is not supported and may cause runtime errors. 

**next.config.js:**
```
const { StylableWebpackPlugin } = require("@stylable/webpack-plugin");
const { attachHook } = require("@stylable/node");

attachHook();

module.exports = {
  webpack: (config) => {
    config.node = {
      fs: "empty",
    };

    safelyWalkJSON(config.module, insertStCssExclude(/\.st\.css$/));

    config.plugins.push(
      new StylableWebpackPlugin({
        includeDynamicModulesInCSS: false,
      })
    );

    return config;
  },
};

function insertStCssExclude(stRegex) {
  return (key, value) => {
    if (value && value.test) {
      if (value.test.toString().includes("css")) {
        if (value.exclude && Array.isArray(value.exclude)) {
          value.exclude.push(stRegex);
        } else if (value.exclude instanceof RegExp) {
          value.exclude = [value.exclude, stRegex];
        } else if (typeof value.exclude === "function") {
          value.exclude = [value.exclude, stRegex];
        } else if (value.exclude) {
          throw new Error("unknown exclude pattern: " + value.exclude);
        } else {
          value.exclude = [stRegex];
        }
      }
    }
  };
}

function safelyWalkJSON(obj, visitor, path = [], visited = new Set()) {
  for (var key in obj) {
    const currentPath = [...path, key];
    if (visited.has(obj[key])) {
      continue;
    } else {
      visited.add(obj[key]);
    }
    const res = visitor(key, obj[key], currentPath);
    if (res === false) {
      continue;
    }
    if (typeof obj[key] === "object") {
      safelyWalkJSON(obj[key], visitor, currentPath, visited);
    }
  }
}
```

**server.js**

```
// server.js
const { createServer } = require("http");
const { parse } = require("url");
const next = require("next");

const dev = process.env.NODE_ENV !== "production";
const app = next({ dev });
const handle = app.getRequestHandler();

require("@stylable/node").attachHook();

app.prepare().then(() => {
  createServer((req, res) => {
    const parsedUrl = parse(req.url, true);

    handle(req, res, parsedUrl);
  }).listen(3000, (err) => {
    if (err) throw err;
    console.log("> Ready on http://localhost:3000");
  });
});
```

**package.json scripts**

```
"scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start"
  }
```

**Comment by barak007** [comment]
Hey 10x for opening an issue. 
Can you provide more details about your setup?
like: 
webpack version
stylable version

or even better link to the project or reproduction repo.

**Comment by ItayTur** [comment]
we talked to vercel and it seems that the issue is steaming from there. 
Currently, there is no support in vercel for nextjs app with custom server, like needed when integrating with stylable.
We decided to use container instead, and deploy it to google cloud. 
We facing new challenges but i will elaborate in a new issue. 
Thx for replying and sorry for the delay.

**Comment by AdamZaczek** [comment]
Thanks for working on this. I was wondering why would stylable not work for me

**Comment by tomrav** [comment]
This appears to be resolved as #1781 reports
